src/FSharp.Core/seq.fs

```diff
@@ -18,6 +18,35 @@ open Microsoft.FSharp.Primitives.Basics
 
 module Internal =
 
+    [<Literal>]
+    let arrayBuilderStartingSize = 4
+
+    [<Struct>]
+    [<NoEquality>]
+    [<NoComparison>]
+    type ArrayBuilder<'T> =
+        {
+            mutable currentCount: int
+            mutable currentArray: 'T array
+        }
+
+    let inline addToBuilder (item: 'T) (builder: byref<ArrayBuilder<'T>>) =
+        match builder.currentCount = builder.currentArray.Length with
```

Comment:
Any reason for `match .. true | false` instead of `if else`?

Reply:
Feels more condensed this way, but I am agnostic.